build: datagen

wawwior opened the pull request:
doing datagen so mojang wont break json schemata.

wawwior committed:
build: datagenned recipes

ThatJadenXgamer commented:
I've never really worked with datagen before and I'm unsure if I will since I find it more reliable to work with than using java classes and build tasks to make jsons

wawwior committed:
build: datagenned advancements

wawwior commented:
You can try it by just checking out the branch and running the `Minecraft Data` task (or `./gradlew runData`) before launching the client! :3
If you build jar files after running `Minecraft Data` or `runData`, the assets will be automatically included there, since `src/generated/resources` is a root directory! :3

The general advantages are, it automatically converts between versions, is much easier to debug and way smaller.

wawwior committed:
build: datagenned archaeology

wawwior committed:
build: (almost) datagenned block loot

wawwior committed:
fix: re-enable other datagen

wawwior committed:
fix: ghost mucus advancement

wawwior committed:
fix: ectoplasm cauldron drops cauldron

wawwior committed:
chore: remove ghost_mucus.json

wawwior committed:
build: datagenned entity loot & brazier chests

wawwior committed:
build: datagenned chest loot

wawwior committed:
chore: formatting

ThatJadenXgamer commented:
this is really interesting, I'll def check it out!

ThatJadenXgamer commented:
also since you've already datagenned the new jsons would it be possible to make another PR real quick with just the new fixed jsons?
it'll save quite a lot of time from updating them, especially the advancements

wawwior committed:
build: datagenned block tags

wawwior committed:
build: datagenned block tags

wawwior committed:
build: datagenned damage type tags

wawwior committed:
build: datagenned entity type tags

wawwior committed:
build: datagenned damage types

wawwior committed:
fix: damage types lookup

wawwior committed:
build: datagenned trim patterns

wawwior committed:
build: datagenned fluid tags

wawwior committed:
build: datagenned data maps

wawwior committed:
build: datagenned item tags

wawwior committed:
fix: vanilla tags

wawwior committed:
build: datagenned biome tags

wawwior committed:
chore: tag organization

wawwior committed:
fix: add has_mega_fossil tag

wawwior committed:
chore: reorganize datagen

wawwior committed:
build: datagenned painting tags

wawwior committed:
chore: reorganize loot table datagen

wawwior committed:
build: datagenned structure tags

wawwior committed:
fix: direct registry lookup

wawwior committed:
chore: move datapack gen

wawwior committed:
build: datagenned configured features

wawwior committed:
build: datagenned jukebox songs

wawwior committed:
build: starting with block state/model datagen [very wip]

BlockFamilies are cool

wawwior committed:
fix: missing tags

wawwior committed:
Revert "build: starting with block state/model datagen [very wip]"

This reverts commit fd09e1ecafa7514f94f9ee315c434d676fd833f1.

wawwior commented:
I decided not to do model generation as I now agree with you on manageability of these...
I might reintroduce `JNEBlockFamilies` later, as broad scale model generation seems viable for those at least, as well as maybe recipe generation.

wawwior committed:
chore: registry consistency

wawwior committed:
chore: commit generated files

wawwior commented:
> also since you've already datagenned the new jsons would it be possible to make another PR real quick with just the new fixed jsons? it'll save quite a lot of time from updating them, especially the advancements

Yup, figured why not commit the generated files; just remember to run datagen before commiting if u change it then :P

wawwior commented:
though honestly im not sure how i feel about commiting all the genned files :/ 
i think it would be cleaner to gitignore them..,,,,

ur call :3

ThatJadenXgamer commented:
it might be a good idea to separate the genned files into a different PR so that it doesn't pull the datagen java files themselves since I would like to mess with them seperately at a later day when I have the time to properly explore it

wawwior commented:
That's gonna break the git history quite a bit and also requires either copying the files to `src/main` or also pulling the updated gradle configuration, but it's possible.
I recommend you checkout this branch, and cherrypick [2933fe0](https://github.com/ThatJadenXgamer/Jadens-Nether-Expansion/pull/236/commits/2933fe002796fea8954a069ebf72544d972e88be) to a new branch from 1.21.1, and make a fixup with the new `build.gradle`

wawwior committed:
docs: javadocs for datagen

wawwior committed:
chore: reformat for readability

ThatJadenXgamer commented:
sure, I'll look into it

wawwior committed:
fix: block loot using wrong registry

wawwior committed:
Revert "chore: commit generated files"

This reverts commit 2933fe002796fea8954a069ebf72544d972e88be.

wawwior committed:
Merge branch '1.21.1' into build/datagen

wawwior committed:
fix: merge commit

wawwior committed:
fix: datagen new tags

wawwior committed:
fix: datagenned vanilla features

wawwior committed:
chore: move datagen out of `providers`

wawwior committed:
build: datagenned configured features

wawwior committed:
Revert "chore: move datagen out of `providers`"

This reverts commit 8975ececff3cb18071cc0b37948ac6f461997f67.

wawwior committed:
build: datagenned biome modifiers

`core/worldgen/feature` -> `core/worldgen/features`
`core/worldgen/feature/config` -> `core/worldgen/features/configuration`

wawwior committed:
Merge branch '1.21.1' into build/datagen

# Conflicts:
#	gradle.properties
#	src/main/java/net/jadenxgamer/netherexp/core/entity/WillOWisp.java
#	src/main/java/net/jadenxgamer/netherexp/core/keys/JNETags.java
#	src/main/java/net/jadenxgamer/netherexp/core/worldgen/features/JNEHugeFungusFeature.java
#	src/main/java/net/jadenxgamer/netherexp/core/worldgen/features/NetherSpeleothemFeature.java
#	src/main/java/net/jadenxgamer/netherexp/registry/JNEFeatureTypes.java
#	src/main/resources/data/minecraft/tags/block/base_stone_nether.json
#	src/main/resources/data/minecraft/tags/block/campfires.json
#	src/main/resources/data/minecraft/tags/block/candles.json
#	src/main/resources/data/minecraft/tags/block/ceiling_hanging_signs.json
#	src/main/resources/data/minecraft/tags/block/climbable.json
#	src/main/resources/data/minecraft/tags/block/combination_step_sound_blocks.json
#	src/main/resources/data/minecraft/tags/block/crops.json
#	src/main/resources/data/minecraft/tags/block/fence_gates.json
#	src/main/resources/data/minecraft/tags/block/fences.json
#	src/main/resources/data/minecraft/tags/block/flower_pots.json
#	src/main/resources/data/minecraft/tags/block/hoglin_repellents.json
#	src/main/resources/data/minecraft/tags/block/ice.json
#	src/main/resources/data/minecraft/tags/block/infiniburn_end.json
#	src/main/resources/data/minecraft/tags/block/infiniburn_nether.json
#	src/main/resources/data/minecraft/tags/block/infiniburn_overworld.json
#	src/main/resources/data/minecraft/tags/block/inside_step_sound_blocks.json
#	src/main/resources/data/minecraft/tags/block/lava_pool_stone_cannot_replace.json
#	src/main/resources/data/minecraft/tags/block/logs.json
#	src/main/resources/data/minecraft/tags/block/mineable/axe.json
#	src/main/resources/data/minecraft/tags/block/mineable/hoe.json
#	src/main/resources/data/minecraft/tags/block/mineable/pickaxe.json
#	src/main/resources/data/minecraft/tags/block/mineable/shovel.json
#	src/main/resources/data/minecraft/tags/block/needs_iron_tool.json
#	src/main/resources/data/minecraft/tags/block/nether_carver_replaceables.json
#	src/main/resources/data/minecraft/tags/block/non_flammable_wood.json
#	src/main/resources/data/minecraft/tags/block/piglin_repellents.json
#	src/main/resources/data/minecraft/tags/block/planks.json
#	src/main/resources/data/minecraft/tags/block/replaceable.json
#	src/main/resources/data/minecraft/tags/block/slabs.json
#	src/main/resources/data/minecraft/tags/block/small_flowers.json
#	src/main/resources/data/minecraft/tags/block/snow_layer_can_survive_on.json
#	src/main/resources/data/minecraft/tags/block/snow_layer_cannot_survive_on.json
#	src/main/resources/data/minecraft/tags/block/soul_fire_base_blocks.json
#	src/main/resources/data/minecraft/tags/block/soul_speed_blocks.json
#	src/main/resources/data/minecraft/tags/block/stairs.json
#	src/main/resources/data/minecraft/tags/block/standing_signs.json
#	src/main/resources/data/minecraft/tags/block/sword_efficient.json
#	src/main/resources/data/minecraft/tags/block/wall_hanging_signs.json
#	src/main/resources/data/minecraft/tags/block/wall_signs.json
#	src/main/resources/data/minecraft/tags/block/walls.json
#	src/main/resources/data/minecraft/tags/block/wither_summon_base_blocks.json
#	src/main/resources/data/minecraft/tags/block/wooden_buttons.json
#	src/main/resources/data/minecraft/tags/block/wooden_doors.json
#	src/main/resources/data/minecraft/tags/block/wooden_fences.json
#	src/main/resources/data/minecraft/tags/block/wooden_pressure_plates.json
#	src/main/resources/data/minecraft/tags/block/wooden_slabs.json
#	src/main/resources/data/minecraft/tags/block/wooden_stairs.json
#	src/main/resources/data/minecraft/tags/block/wooden_trapdoors.json
#	src/main/resources/data/netherexp/worldgen/configured_feature/netherrack_speleothem.json
#	src/main/resources/data/netherexp/worldgen/configured_feature/soul_soil_speleothem.json
#	src/main/resources/data/netherexp/worldgen/placed_feature/netherrack_speleothem.json

wawwior committed:
fix: merge fixup
